tables/basic-table/step8.md

```diff
@@ -0,0 +1,110 @@
+In this step, we'll add in the heart of pagination functionality. We'll add a new state property to track the dynamic rows, and a few more handler functions to facilitate updating the various state properties based off of user interactions.
+
+## Task
+
+1) Create a new state property to represent our dynamic rows, call it “rows”. Set the initial value to the defaultRows constant that you defined earlier.
+
+<pre class="file" data-target="clipboard">
+const [rows, setRows] = React.useState(defaultRows);
+</pre>
+
+2) Locate the code that sets the rows for the table (`rows={defaultRows}`), and update it to reference the new rows state property you just created.
+
+<pre class="file" data-target="clipboard">
+rows={rows}
+</pre>
+
+<strong>Note: </strong> the itemCount prop will still reference the defaultRows constant as itemCount needs to be aware of the total number of rows in the entire dataset, not the number of rows that's being shown at a given time.
+
+3) Add a function “updateList” to handle setting the rows based on when the numPerPage state property changes.
+
+<pre class="file" data-target="clipboard">
+const updateList = value => {
+  setCurrentPage(value);
+  const beginMark = (value - 1) * numPerPage;
+  const endMark = beginMark + numPerPage;
+  let newRows = defaultRows.slice(beginMark, endMark);
+  setRows(newRows);
+};
+</pre>
+
+4) Inside the handlePerPageSelect handler, make a call to updateList and pass it the currentPage state property.
+
+<pre class="file" data-target="clipboard">
+updateList(currentPage);
+</pre>
+
+5) Add an effect that runs when the numPerPage state property changes. Inside this effect, call handlePerPageSelect passing null for the first param (event) and send the numPerPage property as the second param.
+
+<pre class="file" data-target="clipboard">
+React.useEffect(() => {
+  handlePerPageSelect(null, numPerPage);
+}, [numPerPage]);
+</pre>
+
+6) Locate the `<Pagination>` component and set the `onSetPage` prop to an inline arrow function, and within this function make a call to updateList, pass updateList the value you receive as a second parameter.
+
+<pre class="file" data-target="clipboard">
+onSetPage={(_evt, value) => {
+  updateList(value);
+}}
+</pre>
+
+At this point, you should see basic pagination in place. Nice! There is some important cleanup left to do, which we'll get to in the next step.
+
+Your rendered output should look something like the screenshot below;
+
+<img src="basic-table/assets/step-8-complete.png" alt="Image of what table looks like at the end of step 8." style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
+
+Your component implementation should look something like the following;
+
+<pre class="file">
+import * as React from &quot;react&quot;;
+import * as ReactDOM from &quot;react-dom&quot;;
+import { Pagination } from &quot;@patternfly/react-core&quot;;
```

Comment:
Good catch! Actually I had to do this to keep the styles from breaking. I thought this might be a bug at first, but it might be yet another side effect of how our React components strangely auto inject stylesheets.

In the first few steps I only have react-table loaded, not react-core, so I don't get the base styles injected for free. I think the table styles expect base.css to be loaded. At the point that I include react-core in the example, having the entry to base.css seems to cause the css variables to break for the react-core component, so Pagination styles break.

So temporarily I just removed the base style entry once it gets to that point, I figured people might not notice long enough for us to fix it 😅WDYT? 

I wonder if we could simply move [this line](https://github.com/patternfly/training-scenarios/blob/master/react-site/src/App.js#L2) to just before [this one](https://github.com/patternfly/training-scenarios/blob/master/react-site/src/index.js#L3) so that base always comes first? Want me to try it?

Reply:
Nah, we can leave it as is -- might make app.js confusing. I introduced a bug in my example, so that's why I copied the full snippet. The steps are much cleaner now.